### PR TITLE
Addresses #234

### LINKED ISSUE #234: More reshape details
In reshape_diagnosands: Currently the select columns code block is before the name cleaning and so select error message (ie use "bias", "estimand_label" etc) uses the wrong names; also select not being applied at all for cases with no bootstrapping; also cleaning code is there twice.

Order that was there before should be fine:

```
 if (is.null(diagnosis$bootstrap_replicates))  {
  out <- diagnosands_df
  } else {
  out <- reshaping
  }

# Then:
nice_names(out)
# Then:
select(out)
```

Other things:

1. parameters are still being capitalized
2. Too much repetition in se rows: we do not need repetition of parameter names or of N_sims (I also think we we do not need se (bootstrapped) unless there is some other possibility)
3. select should throw an error when only one column is selected

### PR DESCRIPTION
**Pull request by acoppock**
This PR makes it so the select operates on the nice names. closes #234

### CODE REVIEW

**Review by graemeblair — DISMISSED** [review]
LGTM, let's wait for @nfultz review to merge in

File: R/diagnosis_helper_functions.R [hunk 1]
```diff
@@ -154,30 +154,38 @@ reshape_diagnosis <- function(diagnosis, digits = 2, select = NULL) {
...
-  rownames(return_df) <- NULL
+  }
   
+  names_to_change <- setdiff(names(return_df), parameter_names)
+  names(return_df)[names(return_df) %in% names_to_change] <- make_nice_names(names_to_change)
+    
+  # Select columns
+  if (!is.null(select)) {
+    available_to_select <- make_nice_names(c(group_columns, diagnosand_columns))
+    if (any(!(select %in% available_to_select)))
```
Comment (nfultz) [inline R/diagnosis_helper_functions.R]: de morgans law => !all

File: R/diagnosis_helper_functions.R [hunk 2]
```diff
@@ -154,30 +154,38 @@ reshape_diagnosis <- function(diagnosis, digits = 2, select = NULL) {
...
+      stop(paste(
+        "select argument must only include elements from: ",
+        paste(available_to_select, collapse = ", ")
+      ))
```
Comment (nfultz) [inline R/diagnosis_helper_functions.R]: add a test to trigger this stop.

**Review by nfultz — DISMISSED** [review]
lgtm

**Comment by acoppock** [comment]
Thanks, those changes in.

**Review by graemeblair — APPROVED** [review]
good2go

**Pull request merged**